price-feeder/oracle/oracle.go

```diff
@@ -393,6 +411,67 @@ func (o *Oracle) filterDeviations(
 	return filteredPrices, nil
 }
 
+// filterCandleDeviations finds the standard deviations of the tvwaps of
+// all assets, and filters out any providers that are not within 2𝜎 of the mean.
+func (o *Oracle) filterCandleDeviations(
+	candles provider.AggregatedProviderCandles) (
+	provider.AggregatedProviderCandles, error,
+) {
+	var (
+		filteredCandles = make(provider.AggregatedProviderCandles)
+		tvwaps          = make(map[string]map[string]sdk.Dec)
+	)
+
+	for providerName, c := range candles {
+		candlePrices := make(provider.AggregatedProviderCandles)
+
+		for assetName, asset := range c {
+			if _, ok := candlePrices[providerName]; !ok {
+				candlePrices[providerName] = make(map[string][]provider.CandlePrice)
+			}
+			candlePrices[providerName][assetName] = asset
+		}
+
+		tvwap, err := ComputeTVWAP(candlePrices)
+		if err != nil {
+			return nil, err
+		}
+
+		for assetName, asset := range tvwap {
+			if _, ok := tvwaps[providerName]; !ok {
+				tvwaps[providerName] = make(map[string]sdk.Dec)
+			}
+			tvwaps[providerName][assetName] = asset
+		}
+	}
+
+	deviations, means, err := StandardDeviation(tvwaps)
+	if err != nil {
+		return nil, err
```

Comment:
so here we return `nil, err` which is what I'd expect, but for `filterTickerDeviations` we don't seem to do this?